src/HealthCheck.Core/HealthCheckBase.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using HealthCheck.Core.Results;
+
+namespace HealthCheck.Core
+{
+    public abstract class HealthCheckBase : IHealthCheck
```

Comment:
`HealthCheck` is the core component of this library - why would it need to be abstracted? What is `IHealthCheck` and `HealthCheckBase` accomplishing?

Reply:
To have different healthcheck in future. Not insisting. Will fix.